src/entities.rs

```diff
@@ -38,7 +39,7 @@ impl Entity {
     /// `World::spawn_at` for easy serialization.
     pub fn from_bits(bits: u64) -> Self {
         Self {
-            generation: (bits >> 32) as u32,
+            generation: Entities::ret_non_zero_u32((bits >> 32) as u32),
```

Comment:
This is unsound, because `bits >> 32` might be zero. We should use the safe, checked constructor for `NonZeroU32` and modify `from_bits` to return `Option`.